src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -101,6 +102,7 @@ void LRUFileCachePriority::updateSize(int64_t size)
 {
     current_size += size;
     CurrentMetrics::add(CurrentMetrics::FilesystemCacheSize, size);
+    CurrentMetrics::set(CurrentMetrics::FilesystemCacheSizeLimit, getSizeLimit());
```

Comment:
result of `getSizeLimit()` (which is equal to `max_size` in cache config) never changes unless a server is restarted, so no need to do set it again and again here - it can be set in `FileCache's` constructor. 

btw, given that this metric is static - does it make sense to put it in CurrentMetrics? Usually current metrics are dynamic metrics, it seems a bit strange to put never changing value there, but it is ok to do that if it is useful for your usecase.

Reply:
We already have a lot of metrics which "never change", like `HashedDictionaryThreads` or anything related to thread pools. So I think it is Ok.

Reply:
I could think of adding it to some settings table and make it queryable SELECT value clause and convert it to Prometheus metric if needed.

Sometimes we miss some metrics about disk size of SSD ephemeral storages in some cloud providers like AWS. We are setting the cache limit size below max size of the disk. It's slightly different use case but at least it would give some visibility.

Cache limit might change with reboot, moving the server to bigger node host, config change etc. In most cases it will never happen, but publishing current metric and max size will make it easy to calculate current available size and create the alert if cache is full for some period of time

Reply:
Ok, let's just move it as I wrote:
> so no need to do set it again and again here - it can be set only once - in FileCache's constructor.

Reply:
Thanks :+1: 
Moved it to LRUFileCachePriority.h